### ISSUE TITLE
[Feature] Allow a user to backup the files on a Github repository.

### ISSUE THREAD

**Issue by rahulbansal16** [Zettlr/Zettlr#1987]
## Description

I like to use github to store all my codebase. Why can't I use the Zettlr to do the same with my notes.
In this way I will have control over my data and I will be able to browse them through web browser as well.
Zettlr can save my notes automatically on Github on every save

**Comment by nathanlesage** [comment]
Very good idea and this is on the bucket list (git support, that is), but there is no ETA yet.

**Comment by Atulin** [comment]
Here's the activity then, because I came here just to ask for Git integration as well. Opening the folder in Git shell and running `git commit -am "" && git push` isn't all that hard, but it would be nice to have "commit" and "push" buttons directly in the UI.

**Comment by nathanlesage** [comment]
One benefit of the stale bot is that it regularly pulls attention to old issues. Can't give an ETA, but it's on my mind.

**Comment by christopherisnow** [comment]
Here one more voice in the choir :) Has there been any movement on this request since July?

If not, I wonder if anyone has come up with a hack, like running a command from within Zettlr (something like the superb Pandoc integration), or any other workaround? 

FYI I'm personally happy to run Git commands in the terminal, but I'm trying to sell some Word- and Mac-using collaborators on a collective Markdown/Git workflow. For them, doing anything with a CLI could be a dealbreaker, ditto having to open up a separate Git GUI app to save their work to the group Git repo. 

On the other hand, if Zettlr were to have Git GUI support, it would be be the holy grail of plaintext editors for academics. It's pretty close already!

**Comment by nathanlesage** [comment]
@christopherisnow No new movement in this issue per se, but a lot of the code changes since then have been happening with this in mind. It will still take a few months, but then it should go really, really quick.

**Comment by christopherisnow** [comment]
@nathanlesage good news. In the meantime, is it possible to include a git command in the export options as described here? https://github.com/Zettlr/Zettlr/issues/1842

**Comment by nathanlesage** [comment]
As soon as that issue is closed, yup. Will also take a little bit more time, but that issue has priority (b/c they need that for a larger system in which Zettlr will be one part of a larger publishing pipeline)

**Comment by sten0** [comment]
Here's stopgap measure: configure a cron job, systemd timer, or the Windows or macOS equivalent to do something like the following every few minutes:
```
#!/bin/sh
#customise zettle_basedir!
zettlr_basedir="$HOME/Documents/Zettles"
cd "$zettlr_basedir" || exit 1
# Add is needed to add new files
git add * && git commit -m "$(date) AUTO SNAPSHOT" && git push
```
Then use "git rm" to cleanup files (if any were deleted) at the end of the work day.  If all files are unchanged, git does nothing when the timer fires.

AFAIK this is a set it up once and forget about it solution, even on macOS--assuming cron is still part of the default install.  Oh, and here's a crontab snippet (edit the crontab with `crontab -e`)
```
# m h  dom mon dow   command

15  *  *   *   *     /path/to/git/snapshot/script
```

Oh, and I won't recommend GPG signing commits with this method, because the user will become way too relaxed about entering their password after entering it uncountable times every 15 minutes!

**Comment by christopherisnow** [comment]
@nathanlesage just wondering if Git integration has advanced at all. If not, any prognosis?

**Comment by nathanlesage** [comment]
@christopherisnow Zettlr has git detection in the meantime, that is: It checks whether git is installed on the machine, and also indiciates if a directory is also a git repository. I'm sometimes playing around with the git commands to see how they work, but I haven't (yet) progressed in actually implementing these things.

**Comment by christopherisnow** [comment]
@nathanlesage I don't see git detection on 2.0.3. Is this only in the newer (beta?) versions? I wonder how useful this functionality is without the actual git commands and would like to test the waters.

**Comment by nathanlesage** [comment]
That's correct. Git detection only saves it into an environment variable for now. Feel free to have a look into the debug info in the About panel.

The repository detection can be inspected in the properties of a given directory. If it contains a repo, it should show you the git logo.

And yes, it's obviously not useful as of now. But at least the detection is in there.

**Comment by rahulbansal16** [comment]
I feel this could be a big value add if Zettlr can run in the web browser and write directly on the github something like https://pople.app/

**Comment by nathanlesage** [comment]
Definitely, but this is a long-term goal and not in scope for this issue.

**Comment by toidiu** [comment]
First of all thx for making an amazing editor!

I wanted a similar feature and would be very happy if this existed. However, i do think there might be a more extensible feature that would also cover this request.

Instead of `git commit` or something similar could we simply execute an aribitrary shell script instead. Then if we could get hooks at different point (save file, open zettlr editor, open file) that would be huge.

What i am thinking is to use zettlr across multiple machines and make sure it sync to a git repo (pull, push) but the possibilities are endless.
